backend/Testing/Browser/Base/PageTest.cs

```diff
@@ -26,6 +31,21 @@ public PageTest()
     public ILocatorAssertions Expect(ILocator locator) => Assertions.Expect(locator);
     public IPageAssertions Expect(IPage page) => Assertions.Expect(page);
     public IAPIResponseAssertions Expect(IAPIResponse response) => Assertions.Expect(response);
+    /// <summary>
+    /// Consumes a defferred exception that was "thrown" in a sub-thread, and returns it
+    /// or throws if no exception of the given type is found.
+    /// </summary>
+    public Exception ExpectDeferredException<TException>() where TException : Exception
```

Comment:
we might as well return `TException` here as we know that it's that type. That being said, right now you just need to know that `UnexpectedResponseException` is the only type that's supported (though you could just use Exception), is there a better way to do it? Maybe for now we just remove the generic and we can change that later if we have other exception types we care about.

Reply:
OK. I'll limit it for now.

Reply:
this still isn't resolved